### PR TITLE
Schreier-Sims

### PR DESCRIPTION
**Pull request by JannisHeising**

### CODE REVIEW

**Comment by JannisHeising** [comment]
I'm sorry for sending you way too many files again. Maybe next time I'll have that figured out. Also, I don't understand why some tests don't pass here. They all pass on my machine. Is it just version dependent?

File: src/schreier_sims.jl [hunk 1]
```diff
@@ -13,7 +15,7 @@ struct StabilizerChain{T<:AbstractTransversal, P<:AbstractPermutation}
     end
 end
 
-StabilizerChain{T, P}() where {T,P} = StabilizerChain(Vector{T}(), Vector{Vector{P}}())
+StabilizerChain{T, P}() where {T,P} = StabilizerChain(Vector{T}(), Vector{AbstractVector{P}}())
```
Comment (kalmarek) [inline src/schreier_sims.jl:18]: you should never create `Vector{AbstractVector}` on purpose. This should stay as it is, the signature above should be changed. Note that even though `Float64<:Real` we don't have `Vector{Float64} <: Vector{Real}`. We do have `Vector{Float64} <: Vector{<:Real}`

**Review by kalmarek — CHANGES_REQUESTED** [review]

**Comment by kalmarek** [comment]
@JannisHeising the failure is the result of checking in the Manifest. if you click Details to the right of `CI/Julia 1.6 ...` you can see the error message.

I'm not sure how are you using git, but in VSCode you can add files selectively (or even chunks/lines). On the command line you can do `git add src/somefile.jl` to add just one file selectively

**Comment by kalmarek** [comment]
I'm also not sure how do you test (there shouldn't be a need to add `Test` to `Project.toml`).
in julia session while in the correct environment you can either hit `]` to enter pkg-mode and then type `test`, or you can test directly from `julia`-mode by doing `using Pkg; Pkg.test("CGT_UniHeidelberg_2022")`.

So now you need to learn how to rewrite (git) history ;) and remove the unnecessary files from the pull. Use `git rebase -i sha_of_the_last_commit_that_doesn't_need_rewrite`;

Afterwards `git push --force` will forcefully push to tracking branch overriding the history on the server as well. DO NOT use this on branches you don't own, ever ;)

**Comment by JannisHeising** [comment]
I have changed the things you requested (I got a bit stuck on which types I should use and just kept the first thing that worked. Of course it's much better the way you told me). I also fixed the compatibility issues. I mainly use the GitHub Desktop Application, which lets you choose which files you (don't) want to commit. It seems I wasn't careful and committed too much, so now I copied the files I shouldn't have changed from your repo to mine. It's not an elegant solution, but it works.

**Review by kalmarek — APPROVED** [review]

**Pull request merged**